### PR TITLE
Fix duplicated IdC failure metrics

### PR DESCRIPTION
**Pull request by wularr**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Currently, attempting to sign in through the Q by clicking "Use with Pro License" then cancelling results in duplicate browser login and addConnection metrics. This is because we are submitting metrics in the IdC-specific login function and in the general login function (the IdC-specific login function calls the general one).

This change will move all metrics decisions to the IdC-specific login function. The general login function does the actual submission through a hook.

## Checklist
- [X] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [X] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt [hunk 1]
```diff
@@ -145,6 +146,7 @@ abstract class LoginBrowser(
     }
 
     open fun loginIdC(url: String, region: AwsRegion, scopes: List<String>) {
+        val isReAuth = ToolkitConnectionManager.getInstance(project).activeConnection() != null
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:149]: Would this be a proper indicator of reauth, cause this is not feature-specific. So, could there be a scenario where it's authing for one feature but reauthing for another eg in the case of CC in GW and Q in IDE?

Comment (wularr) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt]: New revision inspects scopes to distinguish between Q and non-Q

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt [hunk 2]
```diff
@@ -145,6 +150,19 @@ abstract class LoginBrowser(
     }
 
     open fun loginIdC(url: String, region: AwsRegion, scopes: List<String>) {
+        // assumes scopes contains either Q or non-Q permissions but not both
+        val isReAuth = if (scopes.toSet().intersect(Q_SCOPES.toSet()).isNotEmpty()) {
+            ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(QConnection.getInstance()) != null
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:155]: Should we also compare the start url and region to the active connection's start url and region, cause if they are different then technically it's a new auth?

Comment (manodnyab) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:155]: Can we also move the isReauth logic to a new function?

Comment (wularr) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:155]: > Should we also compare the start url and region to the active connection's start url and region, cause if they are different then technically it's a new auth?

Gonna check in with analytics to see if this is considered new auth. There could be a world where the following can be considered a continuous string of reauths:

User relogs with acc A -> A credentials expire -> user relogs with acc B -> B credentials expire -> user relogs with acc C -> C credentials expire -> ...

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt [hunk 3]
```diff
@@ -110,6 +115,18 @@ abstract class LoginBrowser(
         else -> ""
     }
 
+    private fun isReAuth(scopes: List<String>): Boolean = if (scopes.toSet().intersect(Q_SCOPES.toSet()).isNotEmpty()) {
+        ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(QConnection.getInstance()) != null
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:119]: nit: can we initialize all the connections before so we can reuse the variables?

Comment (wularr) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt]: Pulled `ToolkitConnectionManager.getInstance(project)` to the top. Connections are branch dependent so initializing at the top is undesirable.

**Review by manodnyab — APPROVED** [review]

**Pull request merged**